table/block_based_table_reader.cc

```diff
@@ -208,10 +178,15 @@ class PartitionIndexReader : public IndexReader {
   // return a two-level iterator: first level is on the partition index
   virtual InternalIterator* NewIterator(BlockIter* iter = nullptr,
                                         bool dont_care = true) override {
+    // Filters are already checked before seeking the index
+    const bool skip_filters = true;
+    const bool is_user_data = true;
```

Comment:
yeah I noticed that but I believe mine is more natural. I can still change it back if you feel strongly about it.